src/diagram/models/dq-root.ts

```diff
@@ -31,10 +31,10 @@ export const DQRoot = types.model("DQRoot", {
   }
 }))
 .views(self => ({
-  get reactFlowNodes() {
+  getReactFlowNodes(readOnly?: boolean) {
```

Comment:
@tealefristoe this will decrease the performance because MST won't be able to cache this response anymore. Since this call is probably done many times as nodes are dragged around, it is probably worth fixing this.

There might be a better way, but you could make two getters, one or read only one for not read only.

Reply:
Thanks for catching this. For setting up two getters, could I do something like this?

```getReactFlowNodes(readOnly?: boolean) { ... },
get reactFlowNodes() { return self.getReactFlowNodes(); },
get reactFlowNodesReadOnly() { return self.getReactFlowNodes(true); }```

Or do I need to duplicate the code in `get reactFlowNodes()` in two different getters, like below?

```get reactFlowNodes() {
  ...
  node.getReactFlowNodes(self)
  ...
}

get reactFlowNodesReadOnly() {
  ...
  node.getReactFlowNodes(self, true)
  ...
}```

Reply:
Either should work -- `get` accessor functions without arguments should be automatically cached independent of implementation.